packages/batch/tests/unit/BatchProcessor.test.ts

```diff
@@ -418,4 +426,42 @@ describe('Class: BatchProcessor', () => {
       );
     });
   });
+
+  describe('Batch processing with Lambda context', () => {
```

Comment:
We need addiotnal tests for malformed context, because we cant be certain what the user will pass. I suggest to have tests with empty and malformed options, you can cast it to `as Context`. I'd suggest to have an error message that outputs the context, so the user can inspect it. and copare what was passed to the processor. What do you think?

Reply:
I suppose I may not have full understanding of `Context`, but the batch processing classes are merely passing the `options` passed by a user around, until it can get to the handler on each invocation. But since the handler is user defined, we may not necessarily be able to check when processing is failing directly because of malformed `Context`. I can try adding in a check on required fields and do more research on that side though.

Reply:
Ah, sorry I have poorly phrased my question.

You are absolutely right, we can't asses the core issue with malformed `Context`. 

I want to capture the situation in a test, when user passes the `Context` to the batch processor, but the `Context` is malformed. It can be the user made a mistake, or something happened to the `Context` in between, we don't know. But if this happens, we want to provide a good error message that at shows what was inside the `Context`. This will help the user to troubleshoot the problem further.

Reply:
Exactly, I think it'd be an urge to output and debug the problem. By printing error with the context we can help them to recover.